resources/testdriver.js

```diff
@@ -137,27 +137,64 @@
         }
     };
 
-    window.test_driver_internal = {
+    var manual = {
```

Comment:
I don't quite understand the naming of this, or if the use of `Object.create` with this object as the prototype is a necessary part of the change. Shadowing `in_automation` by setting it to true on the instance (leaving it false on the prototype) works, but why?

Reply:
@jugglinmike ?

Reply:
Originally, we defined the object in this way so could automatically detect
when `test_driver_internal` had been extended (i.e. by comparing the enumerable
properties of `manual` and `test_driver_internal`).

We since decided on a more intentional override mechanism (via the
`in_automation` property), and you are right to observe that the use of the
prototype chain is no longer technically necessary. I've left it in place for
the sake of clarity: it makes the code reflect the intended usage. `manual`
describes an object that exposes the human-driven implementation, and
`test_driver_internal` describes a distinct object which defaults to that
behavior. When/if `test_driver_internal` is overridden, it does not affect the
`manual` implementation conceptually, and the state of the objects reflects
that.

Would naming this object `manuel_implementation` make this any more clear? Or
would you prefer that we simplify the internals instead?

Reply:
I'd prefer to have no separation of the methods that have a default implementation that waits for user input and those that always throw, and overriding all in the same manner. If using the prototype chain and overriding by setting a property shadowing it on the instance is nice, it's equally nice in both cases I think. Just a flag `test_driver_internal` as it originally was would be fine I think.

If one other reviewer does find the new structure clearer then I'll consider myself downvoted, however.

Reply:
@jugglinmike this is the one outstanding thread here; I don't think I have a strong preference either way here, FWIW.

Reply:
I'll merge the PR and send a follow-up.